src/views/language/LanguageVitalityComputation.tsx

```diff
@@ -0,0 +1,155 @@
+import { LanguageData } from '../../types/LanguageTypes';
+
+/**
+ * Maps Ethnologue 2013 vitality levels to 0-9 scale
+ * Ethnologue 2013 scale: 1=National, 2=Regional, 3=Trade, 4=Educational, 5=Written, 6=Threatened, 7=Shifting, 8=Moribund, 9=Nearly Extinct, 10=Extinct
+ * Our scale: 9=National, 8=Regional, 7=Trade, 6=Educational, 5=Written, 4=Threatened, 3=Shifting, 2=Moribund, 1=Nearly Extinct, 0=Extinct
+ */
+export function getEthnologue2013Score(vitality: string): number | null {
+  if (!vitality) return null;
+
+  switch (vitality.toLowerCase()) {
+    case 'national':
+      return 9;
+    case 'regional':
+      return 8;
+    case 'trade':
+      return 7;
+    case 'educational':
+      return 6;
+    case 'written':
+      return 5;
+    case 'threatened':
+      return 4;
+    case 'shifting':
+      return 3;
+    case 'moribund':
+      return 2;
+    case 'nearly extinct':
+      return 1;
+    case 'extinct':
+      return 0;
+    default:
+      return null;
+  }
+}
+
+/**
+ * Maps Ethnologue 2025 vitality levels to 0-9 scale
+ * Ethnologue 2025: Institutional = 9, Stable = 6, Endangered = 3, Extinct = 0
+ */
+export function getEthnologue2025Score(vitality: string): number | null {
+  if (!vitality) return null;
+
+  switch (vitality.toLowerCase()) {
+    case 'institutional':
+      return 9;
+    case 'stable':
+      return 6;
+    case 'endangered':
+      return 3;
+    case 'extinct':
+      return 0;
```

Comment:
Missing numbers -- you'll note if you test this that the 2025 numbers never actually appear.

```suggestion
    case '1 institutional':
      return 9;
    case '2 stable':
      return 6;
    case '3 endangered':
      return 3;
    case '4 extinct':
      return 0;
```